crates/sui-core/src/consensus_adapter.rs

```diff
@@ -737,52 +758,55 @@ impl ConsensusAdapter {
             // We enter this branch when in select above await_submit completed and processed_waiter is pending
             // This means it is time for us to submit transaction to consensus
             let submit_inner = async {
-                let ack_start = Instant::now();
-                let mut retries: u32 = 0;
-                while let Err(e) = self
-                    .consensus_client
-                    .submit_to_consensus(&transactions[..], epoch_store)
-                    .await
-                {
-                    // This can happen during reconfig, or when consensus has full internal buffers
-                    // and needs to back pressure, so retry a few times before logging warnings.
-                    if retries > 30
-                        || (retries > 3 && (is_soft_bundle || !transactions[0].kind.is_dkg()))
-                    {
-                        warn!(
-                            "Failed to submit transactions {transaction_keys:?} to consensus: {e:?}. Retry #{retries}"
-                        );
-                    }
-                    self.metrics
-                        .sequencing_certificate_failures
-                        .with_label_values(&[tx_type])
-                        .inc();
-                    retries += 1;
+                const RETRY_DELAY_STEP: Duration = Duration::from_secs(1);
 
-                    if !is_soft_bundle && transactions[0].kind.is_dkg() {
-                        // Shorter delay for DKG messages, which are time-sensitive and happen at
-                        // start-of-epoch when submit errors due to active reconfig are likely.
-                        time::sleep(Duration::from_millis(100)).await;
-                    } else {
-                        time::sleep(Duration::from_secs(10)).await;
-                    };
+                loop {
+                    // Submit the transaction to consensus and return the submit result with a status waiter
+                    let status_waiter = self
+                        .submit_inner(
+                            &transactions,
+                            epoch_store,
+                            &transaction_keys,
+                            tx_type,
+                            is_soft_bundle,
+                        )
+                        .await;
+
+                    match status_waiter.await {
+                        Ok(BlockStatus::Sequenced(_)) => {
```

Comment:
By having this variant of `BlockStatus`, is the goal to keep track of if a committed transaction is submitted by this validator or another?

Reply:
The objective is not to track on tx level, but rather on block level. If tx gets sequenced via another block (and appears either on consensus output or checkpoint), we are going to continue on the processing anyways (as the future will get aborted) . That simplifies also a bit things as it's an all or nothing retry.